conftest.py

```diff
@@ -111,156 +276,87 @@ def temp_dir():
+            # list
+            ds_fn = request.param[0]
+            opts = request.param[1]
+            cls = None if "cls" not in opts.keys() else opts["cls"]
+            args = None if "args" not in opts.keys() else opts["args"]
+            kwargs = None if "kwargs" not in opts.keys() else opts["kwargs"]
+            dataset = data_dir_load(ds_fn, cls=cls, args=args, kwargs=kwargs)
+        else:
+            dataset = data_dir_load(request.param)
+        return dataset
+    except FileNotFoundError:
+        return pytest.skip(f"Data file: `{request.param}` not found.")
+
+
+@pytest.fixture(scope="class")
+def f(request):
     """
-    if request.cls.__name__ in answer_files:
-        answer_file = answer_files[request.cls.__name__]
-        # Make sure we're not overwriting an existing answer set
-        if os.path.isfile(os.path.join(answer_dir, answer_file)):
-            if request.config.getoption("--answer-store"):
-                raise FileExistsError(
-                    "Error, attempting to overwrite "
-                    "answer file {}. Either specify a new version or "
-                    "set the `--force-override-answers` option".format(answer_file)
-                )
-    else:
-        raise AssertionError
-    request.cls.answer_file = answer_file
+    Fixture for returning the field. Needed because indirect=True is
+    used for loading the datasets.
+    """
+    return request.param
 
 
-def _param_list(request):
-    r"""
-    Saves the non-ds, non-fixture function arguments for saving to
-    the answer file.
+@pytest.fixture(scope="class")
+def d(request):
     """
-    test_params = {}
-    func = request.node.function
-    # co_varnames is all of the variable names local to the function
-    # starting with self, then the passed args, then the vars defined
-    # in the function body. This excludes fixture names
-    args = func.__code__.co_varnames[1 : func.__code__.co_argcount]
-    # funcargs includes the names and values of all arguments, including
-    # fixtures, so we use args to weed out the fixtures. Need to have
-    # special treatment of the data files loaded in fixtures for the
-    # frontends
-    for key, val in request.node.funcargs.items():
-        if key in args and not key.startswith("ds_"):
-            test_params[key] = val
-    # Convert python-specific data objects (such as tuples) to a more
-    # io-friendly format (in order to not have python-specific anchors
-    # in the answer yaml file)
-    test_params = utils._streamline_for_io(test_params)
-    return test_params
+    Fixture for returning the ds_obj. Needed because indirect=True is
+    used for loading the datasets.
+    """
+    return request.param
 
 
-@pytest.fixture(scope="function")
-def hashing(request):
-    r"""
-    Handles initialization, generation, and saving of answer test
-    result hashes.
+@pytest.fixture(scope="class")
+def a(request):
```

Comment:
```suggestion
def axis(request):
```

Reply:
This change (as well as the ones below to `N` and `w`) will require changing the name of the passed fixture in every test that uses it. These fixtures are really only necessary because of the use of `indirect=True` when parameterizing the tests. This was needed in order to get the loading of the datasets correct, if I recall correctly.

Reply:
I can see this will cause a lot of changes, but the issue is that we then end up with tests that look like this
```python
@pytest.mark.usefixtures("hashing")
    @pytest.mark.parametrize("ds", [d9p], indirect=True)
    @pytest.mark.parametrize("f", fields, indirect=True)
    @pytest.mark.parametrize("w", weights, indirect=True)
    @pytest.mark.parametrize("d", objs, indirect=True)
    @pytest.mark.parametrize("a", axes, indirect=True)
    def test_d9p_pixelized_projection_values(self, f, d, a, w, ds):
        [...]
```
which I find extremely hard to read, since the variables aren't self explanatory.

Reply:
+1 @cphyc . Additionally, one-letter variable/function names are virtually impossible to grep for, making them all the more difficult to read.